tests/test_simulator.py

```diff
@@ -0,0 +1,50 @@
+from unittest import TestCase
+
+from unittest.mock import Mock, patch
+
+from roulette import Simulator, Game, Martingale, Table, Wheel
+
+
+class TestSimulator(TestCase):
+    def setUp(self):
+        self.table = Table()
+        self.martingale = Martingale(self.table)
+        self.wheel = Wheel()
+        self.game = Game(self.wheel, self.table)
+        self.simulator = Simulator(self.game, self.martingale)
+
+    def test_gather_appends_max_stake_and_rounds_played_after_one_session(self):
+        session_mock = Mock(name="session_mock", return_value=[1, 2, 5])
+        self.simulator.samples = 1
+
+        with patch("roulette.Simulator.session", session_mock):
+            self.simulator.gather()
+
+        session_mock.assert_called_once()
+
+        expected_value_in_maxima = 5
+        self.assertIn(expected_value_in_maxima, self.simulator.maxima)
+
+        expected_length_in_duration = 3
+        self.assertIn(expected_length_in_duration, self.simulator.durations)
+
+    def test_session_calls_cycle_only_if_players_are_active(self):
```

Comment:
better rename to `test_cycles_if_player_is_active`. session calls cycle is an implementation details.

Reply:
the test can be removed as well as it is obvious cycle will be called when player is active. this test is basically testing while loop (whether loop body is executed when loop condition is True).

Reply:
Removed.